test/engine/SpatialJoinTest.cpp

```diff
@@ -382,77 +386,75 @@ std::vector<std::vector<std::string>> unordered_rows_small{
 // distance from the object to itself should be zero
 std::vector<std::string> expectedDistSelf{"0"};
 
-// distance from Uni Freiburg to Freiburger Müsnster is 2,33 km according to
-// google maps
+// helper functions
 auto P = [](double x, double y) { return GeoPoint(y, x); };
 
-std::vector<std::string> expectedDistUniMun{std::to_string(
-    static_cast<int>(ad_utility::detail::wktDistImpl(P(7.83505, 48.01267),
-                                                     P(7.85298, 47.99557)) *
-                     1000))};
+auto expectedDistBaseline = [](const GeoPoint& p1, const GeoPoint& p2) {
+  return std::to_string(
+      static_cast<int>(ad_utility::detail::wktDistImpl(p1, p2) * 1000));
+};
+
+auto expectedDistS2 = [](const GeoPoint& p1, const GeoPoint& p2) {
+  auto p1_ = S2Point{S2LatLng::FromDegrees(p1.getLat(), p1.getLng())};
+  auto p2_ = S2Point{S2LatLng::FromDegrees(p2.getLat(), p2.getLng())};
+
+  return std::to_string(static_cast<int>(S2Earth::ToMeters(S1Angle(p1_, p2_))));
+};
+
+auto expectedDist = [](const GeoPoint& p1, const GeoPoint& p2) {
+  // TODO<ullingerc> decision baseline vs s2
```

Comment:
Address this TODO

Reply:
The clean solution is to extend the entire tests to always test both `baselineAlgorithm` and `s2geometryAlgorithm`. Since this would mean refactoring significant parts of `SpatialJoinTest.cpp`, I think we should first discuss whether the currently unused `baselineAlgorithm` should be kept.